plugins/woocommerce/client/blocks/assets/js/atomic/blocks/product-elements/button/block.tsx

```diff
@@ -165,20 +164,22 @@ const LoadingAddToCartButton = ( {
 const AddToCartButtonPlaceholder = ( {
 	className,
 	style,
+	blockClientId,
 }: AddToCartButtonPlaceholderAttributes ): JSX.Element => {
-	const blockProps = useBlockProps();
-	const blockClientId = blockProps?.id;
-
 	const {
 		current: currentProductType,
 		registerListener,
 		unregisterListener,
 	} = useProductTypeSelector();
 
 	useEffect( () => {
-		registerListener( blockClientId );
+		if ( blockClientId ) {
```

Comment:
```suggestion
		if ( blockClientId ) {
				registerListener( blockClientId );
				return () => {
						unregisterListener( blockClientId );
				};
		}
```

I prefer this way so the returned callback will be executed only if the blockClientId exists and not always.

Reply:
Good suggestion, merged!